packages/next/src/server/app-render/use-flight-response.tsx

```diff
@@ -38,23 +42,49 @@ export function useFlightStream<T>(
     return response
   }
 
-  // react-server-dom-webpack/client.edge must not be hoisted for require cache clearing to work correctly
-  const { createFromReadableStream } =
-    // eslint-disable-next-line import/no-extraneous-dependencies
-    require('react-server-dom-webpack/client') as typeof import('react-server-dom-webpack/client')
-
-  const newResponse = createFromReadableStream<T>(flightStream, {
-    findSourceMapURL,
-    serverConsumerManifest: {
-      moduleLoading: clientReferenceManifest.moduleLoading,
-      moduleMap: isEdgeRuntime
-        ? clientReferenceManifest.edgeSSRModuleMapping
-        : clientReferenceManifest.ssrModuleMapping,
-      serverModuleMap: null,
-    },
-    nonce,
-    debugChannel: debugStream ? { readable: debugStream } : undefined,
-  })
+  let newResponse: Promise<T>
+  if (flightStream instanceof ReadableStream) {
+    // react-server-dom-webpack/client.edge must not be hoisted for require cache clearing to work correctly
+    const { createFromReadableStream } =
+      // eslint-disable-next-line import/no-extraneous-dependencies
+      require('react-server-dom-webpack/client') as typeof import('react-server-dom-webpack/client')
+
+    newResponse = createFromReadableStream<T>(flightStream, {
+      findSourceMapURL,
+      serverConsumerManifest: {
+        moduleLoading: clientReferenceManifest.moduleLoading,
+        moduleMap: isEdgeRuntime
+          ? clientReferenceManifest.edgeSSRModuleMapping
+          : clientReferenceManifest.ssrModuleMapping,
+        serverModuleMap: null,
+      },
+      nonce,
+      // @ts-expect-error -- when the reactServerStream is a ReadableStream we assume the debugStream is too
+      debugChannel: debugStream ? { readable: debugStream } : undefined,
+    })
+  } else {
+    // react-server-dom-webpack/client.edge must not be hoisted for require cache clearing to work correctly
+    // @ts-expect-error -- node APIs are currently uptyped
+    const { createFromNodeStream } =
+      // eslint-disable-next-line import/no-extraneous-dependencies
+      require('react-server-dom-webpack/client') as typeof import('react-server-dom-webpack/client')
+
+    newResponse = createFromNodeStream<T>(
+      flightStream,
+      {
+        moduleLoading: clientReferenceManifest.moduleLoading,
+        moduleMap: isEdgeRuntime
+          ? clientReferenceManifest.edgeSSRModuleMapping
+          : clientReferenceManifest.ssrModuleMapping,
+        serverModuleMap: null,
+      },
+      {
+        findSourceMapURL,
+        nonce,
+        debugChannel: debugStream,
```

Comment:
doesn't this need a `{ readable: ... }` wrapper around the stream like the branch above?
```suggestion
        debugChannel: { readable: debugStream },
```

Reply:
ah, my bad, the types are different, this is correct
https://github.com/facebook/react/blob/561ee24d4a7c805e9837ee4cfbb5671a35e41b5a/packages/react-server-dom-webpack/src/client/ReactFlightDOMClientNode.js#L65-L67

Reply:
Tangentially related: https://github.com/facebook/react/pull/35039